updating version check to 3.12

sameer-google opened the pull request:
This PR updates the Python version compatibility for gsutil to support Python 3.8 through 3.12, while dropping support for older versions (3.5, 3.6, and 3.7). The following changes have been made:

- Updated version check in `gsutil.py` to `3.8-3.12`.
- Updated the `python_requires` and `classifiers` in `setup.py` to specify compatibility with Python versions 3.8 to 3.12, removing references to unsupported versions (3.5, 3.6, and 3.7).

These updates align the project with the ongoing effort to support the latest Python versions while ensuring compatibility across all tested platforms.

sushantkhanna commented:
Please squash the commits so that only a single commit gets merged into primary branch related to this change.

sushantkhanna commented on setup.py:
```diff
@@ -126,16 +126,16 @@ def make_release_tree(self, base_dir, files):
         'Natural Language :: English',
         'Programming Language :: Python',
         'Programming Language :: Python :: 3',
```
Even this line needs to be removed ?

sameer-google replied:
It's useful for searchability on package indexes like PyPI, as some users may filter for projects compatible with Python 3 broadly, without specifying a minor version. We should only remove this line if the project does not support all minor versions of Python 3. Correct me if i am wrong.

sameer-google replied:
As we are updating `python_requires='>=3.8, <=3.12'`, this ensures that we are compatible only with 3.8 to 3.12.

we can keep the `'Programming Language :: Python :: 3'` line, as long as we are not claiming compatibility with all versions of `Python 3` (which could lead to confusion). IMO, It's fine to leave it, especially if we later specify the exact minor versions (e.g., Programming Language :: Python :: 3.8, Programming Language :: Python :: 3.12) as well.

sameer-google commented:
> Please squash the commits so that only a single commit gets merged into primary branch related to this change. Please update the description.

Squashed the commits and Updated PR description.

sameer-google commented:
The kokoro tests failing in this PR are failing in the previous ones' as well.
The CI / build failures are due to a change in version check. These are resolved by [CI build PR](https://github.com/GoogleCloudPlatform/gsutil/pull/1812), merging alongside this PR.
